pkg/ee/default-policy-controller/controller.go

```diff
@@ -0,0 +1,399 @@
+			binding.Spec = kubermaticv1.PolicyBindingSpec{
+				PolicyTemplateRef: corev1.ObjectReference{
+					Name: template.Name,
+				},
+			}
+
+			logger.Infof("Reconciling PolicyBinding %s/%s", binding.Namespace, binding.Name)
+			return binding, nil
+		}
+	}
+}
+
+// isClusterTargeted checks if the PolicyTemplate targets the given cluster.
+func isClusterTargeted(cluster *kubermaticv1.Cluster, template *kubermaticv1.PolicyTemplate) bool {
+	// If no target is specified, we check the visibility
+	if template.Spec.Target == nil {
+		// Global policies apply to all clusters
+		if template.Spec.Visibility == kubermaticv1.PolicyTemplateVisibilityGlobal {
+			return true
+		}
+
+		// Project policies apply to clusters in the same project
+		if template.Spec.Visibility == kubermaticv1.PolicyTemplateVisibilityProject &&
+			template.Spec.ProjectID != "" &&
+			cluster.Labels[kubermaticv1.ProjectIDLabelKey] == template.Spec.ProjectID {
+			return true
+		}
+
+		return false
+	}
+
+	// Check project selector if specified
+	if template.Spec.Target.ProjectSelector != nil && cluster.Labels[kubermaticv1.ProjectIDLabelKey] != "" {
+		selector, err := metav1.LabelSelectorAsSelector(template.Spec.Target.ProjectSelector)
+		if err != nil {
+			return false
+		}
+
+		projectLabels := map[string]string{kubermaticv1.ProjectIDLabelKey: cluster.Labels[kubermaticv1.ProjectIDLabelKey]}
+		if !selector.Matches(labels.Set(projectLabels)) {
+			return false
+		}
+	}
+
+	// Check cluster selector if specified
+	if template.Spec.Target.ClusterSelector != nil {
+		selector, err := metav1.LabelSelectorAsSelector(template.Spec.Target.ClusterSelector)
+		if err != nil {
+			return false
+		}
+
+		if !selector.Matches(labels.Set(cluster.Labels)) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func enqueueClusters(client ctrlruntimeclient.Client, log *zap.SugaredLogger) handler.EventHandler {
+	return handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, a ctrlruntimeclient.Object) []reconcile.Request {
+		var requests []reconcile.Request
+		policyTemplate := a.(*kubermaticv1.PolicyTemplate)
+
+		// Check if the policy template is enforced
+		if !policyTemplate.Spec.Enforced {
+			return requests
+		}
+
+		// List all clusters
+		clusters := &kubermaticv1.ClusterList{}
+		if err := client.List(ctx, clusters); err != nil {
+			log.Error(err)
+			utilruntime.HandleError(fmt.Errorf("failed to list clusters: %w", err))
+			return requests
+		}
+
+		for _, cluster := range clusters.Items {
+			if isClusterTargeted(&cluster, policyTemplate) {
+				requests = append(requests, reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name: cluster.Name,
+					},
+				})
+			}
+		}
+		return requests
+	})
+}
+
+func withEventFilter() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			obj := e.Object.(*kubermaticv1.PolicyTemplate)
+			if obj.GetDeletionTimestamp() != nil {
+				return false
+			}
```

Comment:
AFAIK it's impossible to create an object with a deletion timestamp already set. This has, once controller-runtime's fake client was improved to reflect this behaviour, caused some of our unit tests to fail in the past when we updating CR.